src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -451,15 +452,14 @@ static std::unique_ptr <STBase> parseLeaf (
 
         try
         {
-            std::unique_ptr <STVector256> tail =
-                std::make_unique <STVector256> (field);
+            STVector256 tail (field);
             for (Json::UInt i = 0; value.isValidIndex (i); ++i)
             {
                 uint256 s;
                 s.SetHex (value[i].asString ());
-                tail->push_back (s);
+                tail.push_back (s);
             }
-            ret = std::move (tail);
+            ret = detail::make_stvar <STVector256> (std::move (tail));
```

Comment:
Premature optimization suggestion:
```
            ret = detail::make_stvar <STVector256> ();
            for (Json::UInt i = 0; value.isValidIndex (i); ++i)
            {
                uint256 s;
                s.SetHex (value[i].asString ());
                ret->push_back (s);
            }
```
Probably not worth it: Removes the move construction and the temp variable from the stack, adds overhead of accessing the `optional` for every entry.